fableous-fe/src/components/canvas/Canvas.tsx

```diff
@@ -566,19 +570,33 @@ const Canvas = forwardRef<HTMLCanvasElement, CanvasProps>(
         default:
           setToolMode(ToolMode.None);
       }
-      const ctx = canvas.getContext("2d");
-      if (ctx) {
-        ctx.textAlign = "center";
-        ctx.textBaseline = "middle";
-      }
-      const ws = wsRef.current;
-      ws?.addEventListener("message", readMessage);
+      wsConn?.addEventListener("message", readMessage);
       return () => {
-        ws?.removeEventListener("message", readMessage);
+        wsConn?.removeEventListener("message", readMessage);
       };
-      // only trigger once during componentMount
+      // only trigger on new websocket connection
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [role, wsConn]);
+
+    // cleanup before moving to next page
+    useEffect(() => {
+      const ctx = canvasRef.current.getContext(
+        "2d"
+      ) as CanvasRenderingContext2D;
+      const { width, height } = canvasRef.current;
+      ctx.clearRect(0, 0, width, height);
+      setAudioB64Strings([]);
+      setTextShapes({});
       // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [role, wsRef]);
+    }, [pageNum]);
+
+    // temporary workaround to recalculate width when canvas appears or becomes hidden
+    useEffect(() => {
+      const canvas = canvasRef.current;
+      canvas.width = canvas.offsetWidth * SCALE;
```

Comment:
extract this canvas resize and onmount useEffect to a separate func?

Reply:
done